source/Halibut/Diagnostics/HalibutTimeoutsAndLimits.cs

```diff
@@ -153,7 +153,16 @@ public TimeSpan SafeTcpClientPooledConnectionTimeout
         /// The duration a TCP connection will wait for a keepalive response before sending another keepalive probe.
         /// </summary>
         public TimeSpan TcpKeepAliveInterval { get; set; } = TimeSpan.FromSeconds(5);
-        
+
+        /// <summary>
+        /// The maximum number of authorised TCP connections per thumbprint. Used to prevent denial-of-service/connection exhaustion due to too many incoming connections from a single client
+        /// </summary>
+        /// <remarks>
+        /// Halibut is not sentient
+        /// Applies in both directions //TODO - Be more verbose
+        /// </remarks>
+        public int MaximumAuthorisedTcpConnectionsPerThumbprint { get; set; } = int.MaxValue;
```

Comment:
I know this a draft PR  and this is just for completeness:
- Default to 10
-  Allow to override the limit via an environment variable

Reply:
What the comment is yet to cover is that this also limits the number of connections made to a listening tentacle, setting this to 10 means octopus can make up to 10 connections to a DW for example.

Reply:
Good point. Let's make it specific to the polling mode.

Reply:
Changed in [8d40246](https://github.com/OctopusDeploy/Halibut/pull/599/commits/8d402467b99c0d0fe7fcdb92b6c584232a3153c6) to be polling subscriptions only